pkg/orchestrator/orchestrator.go

```diff
@@ -31,67 +28,86 @@ const (
 	BuildReasonUnknown         = "Unknown"
 )
 
-type OrchestratorSpecificConfigProviding interface {
-	InitOrchestratorProvider(settings *OrchestratorSettings)
+var (
+	provider     ConfigProvider
+	providerOnce sync.Once
+)
+
+type ConfigProvider interface {
+	Configure(opts *Options) error
 	OrchestratorType() string
 	OrchestratorVersion() string
-	GetStageName() string
-	GetBranch() string
-	GetReference() string
-	GetBuildURL() string
-	GetBuildID() string
-	GetJobURL() string
-	GetJobName() string
-	GetCommit() string
-	GetPullRequestConfig() PullRequestConfig
-	GetRepoURL() string
+	StageName() string
+	Branch() string
+	GitReference() string
+	RepoURL() string
+	BuildURL() string
+	BuildID() string
+	BuildStatus() string
+	BuildReason() string
+	JobURL() string
+	JobName() string
```

Comment:
These renamings are for readability first. Because defining getters without `Get` prefix in method name is idiomatic way of naming in Go: https://go.dev/doc/effective_go#Getters